CONTRIBUTING.md

```diff
@@ -11,6 +11,38 @@ and smoother if you have the time and energy to read and follow these suggestion
 
 There's never enough!  Pretty much anything is fair game to improve here.
 
+### Running the doc build yourself
+
+To check your documentation additions, you can optionally build the docs yourself and see how the changes will look on the webpage. 
+
+To run the doc build yourself, the setup step is:
+```
+$ cargo install --vers "^0.4" mdbook
+$ cargo install mdbook-linkcheck
+$ cargo install mdbook-mermaid
+$ cargo install gelatyx --version "^0.2"
+```
+Then build the docs with:
+```
+$ ./ci/build-docs.sh
+```
+And open them in your browser with (macOS):
+```
+$ open gh_pages/index.html
+```
+Or (Linux):
+```
+$ google-chrome gh_pages/index.html
+```
+Or (Linux):
+```
+$ firefox gh_pages/index.html
+```
+
+
+
+
+
```

Comment:
You may wish to break the command part of this out in a separate block; due to the way that suggestions work in GH, it's not possible to insert blocks in a suggestion block:

```suggestion
To serve them up and automatically rebuild and refresh the docs in your browser, run `ci/build-docs.sh serve` and then click on the URL that it prints out after it has performed the first build.

Any arguments passed to `build-docs.sh` are passed down to the underlying `mkdocs` utility.

Look at [mkdocs serve](https://www.mkdocs.org/user-guide/cli/#mkdocs-serve) for more information on additional parameters.
```